src/compiler/checker.ts

```diff
@@ -41224,7 +41224,7 @@ namespace ts {
                 if (symbol && (symbol === undefinedSymbol || symbol === globalThisSymbol || symbol.declarations && isGlobalSourceFile(getDeclarationContainer(symbol.declarations[0])))) {
                     error(exportedName, Diagnostics.Cannot_export_0_Only_local_declarations_can_be_exported_from_a_module, idText(exportedName));
                 }
-                else {
+                else if (!node.parent.parent.isTypeOnly) {
                     markExportAsReferenced(node);
                     const target = symbol && (symbol.flags & SymbolFlags.Alias ? resolveAlias(symbol) : symbol);
                     if (!target || target === unknownSymbol || target.flags & SymbolFlags.Value) {

```

Comment:
1. We shouldn’t be skipping the rest of this block (skipping `checkExpressionCached` could leave a completely invalid RHS unchecked).
2. `markExportAsReferenced` already has code to skip marking something as referenced if it is or resolves to a type-only declaration. I’m honestly not super familiar with what happens in `transpileModule`, but I’m guessing it gets resolved to `unknownSymbol` and therefore never checks? At any rate, it seems like that would be the place to make the change. Can you check and see if that makes sense?

Reply:
If we allow `checkExpressionCached` to continue, the import equals declaration still ends up marked referenced via `checkExpressionCached` -> `checkIdentifier` -> `markAliasReferenced` -> `markAliasSymbolAsReferenced`. In other words, it's not only `markExportAsReferenced` that will end up marking it referenced.

So I think it's possible to alter `markExportAsReferenced` instead of `checkExportSpecifier`, though it may be a bit more awkward since `checkExportSpecifier` takes a couple different kinds of nodes.

And then we still need to handle the other code path too. For that, nothing stands out as a great option. It feels like an `export type` ancestor check is possible, and similar to other checks that `markAliasReferenced` does already for example (`isInTypeQuery`), but I'm not sure if there are performance concerns with that here.

Reply:
I made the changes I described above, let me know if a different approach would be preferable. Thanks!